src/ocean/io/device/File.d

```diff
@@ -412,6 +412,18 @@ class File : Device, Device.Seek, Device.Truncate
                 return idup(path_);
         }
 
+        /**********************************************************************
+
+            Returns:
+                path used by this file.
+
+        **********************************************************************/
+
+        public cstring path ()
+        {
+            return this.path_;
+        }
```

Comment:
Why isn't this done in `toString()` too?

Reply:
Because `toString()` must return `istring` (from `Object`).